### PR TITLE
[stable9.1] Additional perm check in Webdav

### PR DESCRIPTION
**Pull request by PVince81**
Backport of https://github.com/owncloud/core/pull/25388 to stable9.1

Please review @SergioBertolinSG @owncloud/filesystem @DeepDiver1975 @georgehrke

### CODE REVIEW

**Comment by PVince81** [comment]
I did a manual retest with cadaver and it still works.

**Comment by guruz** [comment]
👍

File: apps/dav/lib/Connector/Sabre/ObjectTree.php [hunk 1]
```diff
@@ -283,6 +283,11 @@ public function copy($source, $destination) {
 			throw new InvalidPath($ex->getMessage());
 		}
 
+		// Webdav's copy will implicitly do a delete+create, so only create+delete permissions are required
+		if (!$this->fileView->isCreatable($destinationDir)) {
```
Comment (DeepDiver1975) [inline apps/dav/lib/Connector/Sabre/ObjectTree.php]: ObjectTree is only use in v1 webdav endpoint - we need to have a look with respect to this behavior on the v2 endpoint

@SergioBertolinSG can I ask you to have a look at the integration test to address remore.php/dav?

THX

Comment (PVince81) [inline apps/dav/lib/Connector/Sabre/ObjectTree.php]: Indeed. Note that I had a quick manual test against the new endpoint and the behavior was correct.
But would be good to have an integration test for this one too.

Comment (SergioBertolinSG) [inline apps/dav/lib/Connector/Sabre/ObjectTree.php]: Sure, I can add it in another PR if you need this merged quickly.

Comment (PVince81) [inline apps/dav/lib/Connector/Sabre/ObjectTree.php]: @DeepDiver1975 would be good to merge this one quickly

Comment (PVince81) [inline apps/dav/lib/Connector/Sabre/ObjectTree.php]: @SergioBertolinSG please do, as separate PR against master, which we will then backport

**Pull request merged**